apps/chef_objects/src/chef_objects.app.src

```diff
@@ -29,5 +29,71 @@
                   folsom,
                   pooler
                  ]},
-  {env, []}
+  {env, [
+         %% S3 Access credentials for Bookshelf operations
+         {s3_access_key_id, undefined},
+         {s3_secret_key_id, undefined},
+
+         %% s3_url is the endpoint used by the chef-client for
+         %% retreiving cookbooks
+         {s3_url, undefined},
+
+         %% s3_external_url is the endpoint used by Chef Server for
+         %% retreiving cookbooks
+         {s3_external_url, undefined},
+
+         %% The name of the Amazon S3 bucket. (This is configurable
+         %% and may be pointed at external storage locations, such as
+         %% Amazon EC2. Four settings require configuration when
+         %% pointing at an external cookbook storage location:
+         %% bookshelf['vip'], for the endpoint used by the
+         %% chef-client; bookshelf['external_url'], for the endpoint
+         %% used by the Chef server; bookshelf['access_key_id'], for
+         %% user access key; bookshelf['secret_access_key'], for
+         %% secret key; and opscode_erchef['s3_bucket'], for the
+         %% bucket name. Reconfigure the Chef server after changing
+         %% these settings
+         {s3_platform_bucket_name, "bookshelf"},
+
+
+         %% Timeout value in milliseconds for S3 operations
+         {s3_parallel_ops_timeout, 5000},
+
+         %% The number of processes available to run parallel S3
+         %% operations
+         {s3_parallel_ops_fanout, 20},
+
+         %% The amount of time (in milliseconds) to wait for cookbook
+         %% dependency problems to be solved.
+         {depsolver_timeout, 5000},
+
+         %% Use to specify the amount of time (in seconds) before
+         %% connections to the server expire. If node bootstraps are
+         %% timing out, increase this setting. Default value: 28800.
+         {s3_url_ttl, 28800},
```

Comment:
I think the description on this one is misleading.  Before this new feature I would have described this as "The amount of time a signed s3_URL generated by the Chef server will be valid.  After this timeout, requests using the link will fail with HTTP 403".